ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceAlgorithms.java

```diff
@@ -0,0 +1,136 @@
+package com.microsoft.azure.kusto.ingest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.kusto.data.Utils;
+import com.microsoft.azure.kusto.data.instrumentation.FunctionOneException;
+import com.microsoft.azure.kusto.data.instrumentation.MonitoredActivity;
+import com.microsoft.azure.kusto.data.instrumentation.Tracer;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException;
+import com.microsoft.azure.kusto.ingest.resources.RankedStorageAccount;
+import com.microsoft.azure.kusto.ingest.resources.ResourceWithSas;
+import com.microsoft.azure.kusto.ingest.utils.SecurityUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ResourceAlgorithms {
```

Comment:
I'm sure I'm just missing something about the rationale for this design, but as far as I can tell, `ResourceAlgorithms` provides methods that I would expect to be a part of either `ResourceManager`, or some parent or inherited class that exposes the subset of methods that should be exposed publicly.

I generally regard (effectively) static classes as a smell - `resourceActionWithRetries`, `postToQueueWithRetries`, `uploadStreamToBlobWithRetries` and `uploadLocalFileWithRetries` accept a `ResourceManager`, and based on these methods' names, at first glance I'd expect them to be called thusly, for example: `resourceManager.postToQueueWithRetries(...)`

As for the remaining methods, `roundRobinNestedList`, `getShuffledResources` and `groupResourceByAccountName`, I'd expect them to be in a `ResourceGroup<T>` class.

Again, I assume I'm misunderstanding something, but at first glance the design of this class is a smell to me.

Reply:
They were, @yogilad wanted me to move them because they "bloated the ResourceManager class".

I'm not sure what to do when people have radically different ideas of arranging code.

Maybe we can all discuss together.